optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/PlannerBenchmarkRunner.java

```diff
@@ -169,13 +172,41 @@ private void warmUp() {
         for (SolverBenchmarkResult solverBenchmarkResult : plannerBenchmarkResult.getSolverBenchmarkResultList()) {
             TerminationConfig originalTerminationConfig = solverBenchmarkResult.getSolverConfig().getTerminationConfig();
-            if (!originalTerminationConfigMap.containsKey(solverBenchmarkResult)) {
-                originalTerminationConfigMap.put(solverBenchmarkResult, originalTerminationConfig);
+            ConfigBackup configBackup = new ConfigBackup(originalTerminationConfig);
+            for (SingleBenchmarkResult singleBenchmarkResult : solverBenchmarkResult.getSingleBenchmarkResultList()) {
+                List<PureSingleStatistic> originalPureSingleStatisticList = singleBenchmarkResult.getPureSingleStatisticList();
+                List<PureSingleStatistic> singleBenchmarkStatisticPutResult = configBackup.getPureSingleStatisticMap().put(singleBenchmarkResult, originalPureSingleStatisticList);
+                if (singleBenchmarkStatisticPutResult != null) {
+                    throw new IllegalStateException("SingleBenchmarkStatisticMap of ConfigBackup ( " + configBackup
+                            + " ) already contained key ( " + singleBenchmarkResult + " ) with value ( "
+                            + singleBenchmarkStatisticPutResult + " ).");
+                }
+                singleBenchmarkResult.setPureSingleStatisticList(Collections.<PureSingleStatistic>emptyList());
+
+                ProblemBenchmarkResult problemBenchmarkResult = singleBenchmarkResult.getProblemBenchmarkResult();
+                List<ProblemStatistic> originalProblemStatisticList = problemBenchmarkResult.getProblemStatisticList();
+                if (!originalProblemStatisticMap.containsKey(problemBenchmarkResult)) { // TODO: After Java 8, do Map#putIfAbsent
+                    List<ProblemStatistic> problemStatisticPutResult = originalProblemStatisticMap.put(problemBenchmarkResult, originalProblemStatisticList);
+                    if (problemStatisticPutResult != null && !originalProblemStatisticList.isEmpty()) {
+                        throw new IllegalStateException("OriginalProblemStatisticMap already contained key ( "
+                                + problemBenchmarkResult + " ) with value ( "
+                                + problemStatisticPutResult + " ).");
+                    }
+                }
+                singleBenchmarkResult.getProblemBenchmarkResult().setProblemStatisticList(Collections.<ProblemStatistic>emptyList());
+                singleBenchmarkResult.initSingleStatisticMap();
+            }
+            ConfigBackup configBackupPutResult = configBackupMap.put(solverBenchmarkResult, configBackup);
+            if (configBackupPutResult != null) {
+                throw new IllegalStateException("ConfigBackupMap already contained key ( " + solverBenchmarkResult
+                        + " ) with value ( " + configBackupPutResult + " ).");
             }
```

Comment:
Can we move the whole backup and restoring logic into a method in that new inner class (currently called BackupConfig)?